src/connector/src/source/kafka/enumerator/client.rs

```diff
@@ -32,6 +33,14 @@ use crate::source::kafka::{
 };
 use crate::source::SourceEnumeratorContextRef;
 
+pub static SHARED_KAFKA_CLIENT: LazyLock<tokio::sync::Mutex<HashMap<u64, SharedKafkaItem>>> =
```

Comment:
```suggestion
pub static SHARED_KAFKA_CLIENT: LazyLock<tokio::sync::Mutex<HashMap<KafkaConnection, SharedKafkaItem>>> =
```

Please just use the `KafkaConnection` as the key. The `HashMap` is supposed to compute the hash code and avoid collision.